src/components/BackToTopButton/BackToTopButton.svelte

```diff
@@ -0,0 +1,158 @@
+<script>
+	import { onMount } from 'svelte';
+
+	// All sizing props can be passed with or without 'px' unit.
+	// For example, '40px' and '40' are both valid values.
+	export let size = '40px';
+	export let padding = '12px';
+	export let color = '';
+	export let bgColor = '';
+	export let border = '1px';
+	export let borderRadius = '2px';
+	export let margin = '20px';
+	// The button will be visible when the user scrolls down
+	// by 'showAt' value.
+	export let showAt = 50;
+
+	// Function for auto appending 'px' unit.
+	function appendPx(value = '') {
+		value = value.toLowerCase();
+		if (
+			value.endsWith('%') ||
+			value.endsWith('vh') ||
+			value.endsWith('vw') ||
+			value.endsWith('em') ||
+			value.endsWith('rem') ||
+			value.endsWith('ex') ||
+			value.endsWith('ch') ||
+			value.endsWith('vmin') ||
+			value.endsWith('vmax') ||
+			value.endsWith('cm') ||
+			value.endsWith('mm') ||
+			value.endsWith('in') ||
+			value.endsWith('pt') ||
+			value.endsWith('pc')
+		) {
+			return value;
+		}
+		if (!value.endsWith('px')) {
+			return value + 'px';
+		}
+		return value;
+	}
+
+	// Auto append 'px' unit for all props if not provided.
+	size = appendPx(size);
+	padding = appendPx(padding);
+	border = appendPx(border);
+	borderRadius = appendPx(borderRadius);
+	margin = appendPx(margin);
+
+	let scale = 1;
+	try {
+		// Parse float using substring to remove 'px' unit.
+		scale = parseFloat(size.substring(0, size.length - 2)) / 40;
+	} catch (e) {
+		// Do nothing.
+	}
+
+	// (Default) minus value means the user has not scrolled yet.
+	// So, the button will be initially shown.
+	let scrollY = -1;
+	let show = true;
+
+	function onScroll() {
+		// Set 'scrollY' value.
+		scrollY = window.scrollY;
+
+		// Only show button when the user has scrolled down
+		// by 'showAt' value.
+		show = scrollY >= 0 && scrollY >= showAt;
+	}
+
+	function onClick() {
+		window.scrollTo({ top: 0, behavior: 'smooth' });
+	}
+
+	onMount(() => {
+		window.addEventListener('scroll', onScroll);
```

Comment:
Svelte recommends binding the event to the window using `<svelte:window />` since it will handle the event cleanup as well. [See the example](https://svelte.dev/examples/svelte-window)

Reply:
I've tried this one but it's not working. So, I switched to native js. By the way, I'll look into it again krub :).

Reply:
Interesting... Maybe they don't support all the native events yet. Then that's fine. I suggest adding `removeEventListener` in Svelte's `onDestroy` function to manually remove the event handler when the component leaves the DOM.

Reply:
Just tried it again krub. It seems that `<svelte:window bind:scrollY={scrollY}>` is not working. So, I use `onDestroy` pattern as you suggested.